packages/modals/src/elements/TooltipModal/TooltipModal.tsx

```diff
@@ -130,7 +145,7 @@ export const TooltipModal = React.forwardRef<HTMLDivElement, ITooltipModalProps>
       getModalProps
     } = useModal({
       id,
-      onClose,
+      onClose: handleClose,
```

Comment:
For `TooltipModals` that are animated, the `onClose` should invoke asynchronously — after the animation completes. This will prevent UI that is determined by the state that's changed in the `onClose` from being out of sync.